app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -93,6 +95,13 @@ protected void onCreate(final Bundle savedInstanceState) {
         } catch (IOException e) {
             LOGGER.info("Bluetooth handler not started, most likely Bluetooth is not enabled");
             this.enableBluetooth(false);
+        } catch (NullPointerException e) {
+            LOGGER.error("Could not start the Bluetooth service", e);
+            new SweetAlertDialog(this, SweetAlertDialog.WARNING_TYPE)
+                    .setTitleText(this.getString(R.string.something_went_wrong))
+                    .setContentText(this.getString(R.string.something_went_wrong_QR_popup_explanation))
```

Comment:
OK, but "Something went wrong while scanning the QR code, please try again." doesn't seem to be the message you want to display at that point in time, right? Its just the Bluetooth service that is `null`?